parsing/tok.ml

```diff
@@ -17,7 +17,7 @@ type t =
   | PATTERNIDENT of string
   | IDENT of string
   | FIELD of string
-  | INT of string
+  | NUMERAL of string
```

Comment:
I would strongly prefer if this was a properly typed record with the components of the numeral token.

Reply:
Unfortunately, at this point (in the lexer) it is not known yet whether this token will be parsed as a number or as a user defined notation. So, if we decompose the string in a record there, we must either:
1. keep the original string along the record;
1. or be able to retrieve it from the record.

(1.) means data duplication whereas (2.) means more code, hence potentially more bugs (in particular, one is tempted to normalize the input but if "00" is lexed as "0", this can break a potential user defined "00" notation).

Thus, it seems both easier and safer to delay the spliting of the different components of the token in interp/notation.ml when we know we are actually parsing it as a number.

Reply:
That kind of ambiguity at the lexing level seems indeed to create many problems for little gain. What notations would we lose if we unambiguously parse numerals [thus making them not available to notations]

Reply:
I agree but at least "0" and "1" seem commonly used.

Reply:
Maybe it would less expensive still to have the order emit a num token in that case but have the notations code match it for that cases?

Reply:
Sorry, I'm not sure to fully grasp your idea. Do you offer to separate numerals and notations except for "0" and "1"? How would that be "less expensive" than the current implementation?

Reply:
@ejgallego: I tried to split the components of the numeral token in a record but it seems that coqpp is only designed to hold strings in tokens. So this would imply heavy modifications to coqpp and the result seems not worth all the trouble.